syntax = "proto3";

package filter;

// Сервис фильтрации контента
service FilterService {
  // Фильтрация одного текста
  rpc FilterText(FilterRequest) returns (FilterResponse);
  
  // Батч фильтрация
  rpc FilterBatch(FilterBatchRequest) returns (FilterBatchResponse);
}

// Pre-Filter сервис  
service PreFilterService {
  rpc Predict(PreFilterRequest) returns (PreFilterResponse);
}

// Safety-LLM сервис
service SafetyLLMService {
  rpc Analyze(SafetyLLMRequest) returns (SafetyLLMResponse);
}

// Основной запрос фильтрации
message FilterRequest {
  string text = 1;
  string llm_id = 2;
  string request_id = 3;
  map<string, string> metadata = 4;
}

message FilterResponse {
  string status = 1;          // "safe" | "unsafe"
  string category = 2;        // категория опасности
  string comment = 3;         // пояснение
  int32 processing_ms = 4;    // время обработки
  string request_id = 5;
}

// Батч запрос
message FilterBatchRequest {
  repeated FilterRequest requests = 1;
}

message FilterBatchResponse {
  repeated FilterResponse responses = 1;
}

// Pre-Filter запрос/ответ
message PreFilterRequest {
  string text = 1;
  string request_id = 2;
}

message PreFilterResponse {
  bool safe = 1;
  bool uncertain = 2;
  float score = 3;
  string request_id = 4;
}

// Safety-LLM запрос/ответ
message SafetyLLMRequest {
  string text = 1;
  string request_id = 2;
  float pre_score = 3;
}

message SafetyLLMResponse {
  string status = 1;      // "safe" | "unsafe"
  string category = 2;    // категория
  string reasoning = 3;   // обоснование
  float confidence = 4;   // уверенность модели
  string request_id = 5;
} 